jetty-servlet/src/test/java/org/eclipse/jetty/servlet/ServletContextHandlerTest.java

```diff
@@ -724,6 +724,39 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IO
         assertEquals("getContextPath()=[]", response.getContent(), "response content");
     }
 
+    /**
+     * Address spec "3.5. Request Path Elements" with respect to Servlet Path.
+     */
+    @ParameterizedTest
+    @ValueSource(strings = {"/*", "/", ""})
+    public void testGetServletPathEmpty(String inputServletPath) throws Exception
+    {
+        ServletContextHandler contextHandler = new ServletContextHandler();
+        contextHandler.setContextPath("");
+        contextHandler.addServlet(new ServletHolder(new HttpServlet()
+        {
+            @Override
+            protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException
+            {
+                resp.setContentType("text/plain");
+                resp.setCharacterEncoding("utf-8");
+                resp.getWriter().printf("getServletPath()=[%s]", req.getServletPath());
+            }
+        }), inputServletPath);
+        _server.setHandler(contextHandler);
+        _server.start();
+
+        StringBuilder rawRequest = new StringBuilder();
+        rawRequest.append("GET / HTTP/1.1\r\n");
+        rawRequest.append("Host: local\r\n");
+        rawRequest.append("Connection: close\r\n");
+        rawRequest.append("\r\n");
+        String rawResponse = _connector.getResponse(rawRequest.toString());
+        HttpTester.Response response = HttpTester.parseResponse(rawResponse);
+        assertEquals(200, response.getStatus(), "response status");
+        assertEquals("getServletPath()=[]", response.getContent(), "response content");
```

Comment:
This test is not correct for pathSpec "/", which is the default match and the servletPath should be the entire path, which in this case is "/".
As per the linked document:
>  in the case where the request is matched with the "/*" or "" pattern, in which case it is an empty string
So it is only those patterns that should pass this test.